source/slang/hlsl.meta.slang

```diff
@@ -9894,15 +9894,23 @@ vector<T,N> fmax3(vector<T,N> x, vector<T,N> y, vector<T,N> z)
 
 // minimum
 __generic<T : __BuiltinIntegerType>
-__target_intrinsic(hlsl)
-__target_intrinsic(glsl)
-__target_intrinsic(metal)
-__target_intrinsic(cuda, "$P_min($0, $1)")
-__target_intrinsic(cpp, "$P_min($0, $1)")
-__target_intrinsic(spirv, "OpExtInst resultType resultId glsl450 fus(FMin, UMin, SMin) _0 _1")
 [__readNone]
 [require(cpp_cuda_glsl_hlsl_metal_spirv, sm_2_0_GLSL_140)]
-T min(T x, T y);
+T min(T x, T y)
+{
+    __target_switch
+    {
+    case hlsl:
+    case glsl:
+    case metal:
+        __intrinsic_asm "min";
+    case cuda:
+    case cpp:
+        __intrinsic_asm "$P_min($0, $1)";
+    case spirv:
+        __intrinsic_asm "OpExtInst resultType resultId glsl450 fus(FMin, UMin, SMin) _0 _1";
+    }
+}
```

Comment:
This change may trigger an unexpected issue described in #4093
https://github.com/shader-slang/slang/issues/4093
If CI fails, try again without this change.

Reply:
That's really weird, they should generate the same spirv.

Reply:
This is fixed. The reason is that if there is a global `static const int x = min(1,2)`, we will end up with an `SPIRVAsm` inst in global scope and our spirv emit logic can't handle that.

The fix is to make sure we inline all global scope spirvasms into their use sites.

Reply:
Interesting that 4093 is also resolved...
I am not sure what the cause was but I am happy not to worry about it anymore.
Great job.